### PR TITLE
easypdkprog: init at 1.2

### PR DESCRIPTION
**Pull request by david-sawatzke**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

The udev rules stuff is taken from the openocd package, but I'm not sure if it works

### CODE REVIEW

File: pkgs/development/tools/misc/easypdkprog/default.nix
Comment (danieldk) [inline pkgs/development/tools/misc/easypdkprog/default.nix:10]:
```suggestion
    rev = version;
```

File: pkgs/development/tools/misc/easypdkprog/default.nix
Comment (danieldk) [inline pkgs/development/tools/misc/easypdkprog/default.nix:16]:
```suggestion
    install -Dm755 -t $out/bin easypdkprog
```

File: pkgs/development/tools/misc/easypdkprog/default.nix
Comment (danieldk) [inline pkgs/development/tools/misc/easypdkprog/default.nix:21]:
```suggestion
    install -Dm644 -t $out/etc/udev/rules.d udevrules/70-stm32vcp.rules
```

File: pkgs/development/tools/misc/easypdkprog/default.nix
Comment (danieldk) [inline pkgs/development/tools/misc/easypdkprog/default.nix:25]:
```suggestion
    description = "Read, write and execute programs on PADAUK microcontroller";
```

File: pkgs/development/tools/misc/easypdkprog/default.nix
Comment (danieldk) [inline pkgs/development/tools/misc/easypdkprog/default.nix:26]:
```suggestion
    homepage = "https://free-pdk.github.io/";
```

Or perhaps: https://github.com/free-pdk/easy-pdk-programmer-software

File: pkgs/development/tools/misc/easypdkprog/default.nix
Comment (danieldk) [inline pkgs/development/tools/misc/easypdkprog/default.nix:27]: I think this should be

```suggestion
    license = licenses.gpl3Plus;
```

I looked at some source files and they said GPLv3 or later.

File: pkgs/development/tools/misc/easypdkprog/default.nix
Comment (danieldk) [inline pkgs/development/tools/misc/easypdkprog/default.nix:28]: Add yourself as a maintainer to `maintainers/maintainer-list.nix` . We probably do not want to add a derivation, unless it has a maintainer (or is very popular).

**Review by danieldk — APPROVED** [review]
Thanks for contributing this derivation! I have added a bunch of suggestions for improvements.

File: pkgs/development/tools/misc/easypdkprog/default.nix
Comment (david-sawatzke) [reply pkgs/development/tools/misc/easypdkprog/default.nix]: Whoops, thanks for noticing

**Comment by david-sawatzke** [comment]
@danieldk Thanks for the review, should be fine now

File: pkgs/development/tools/misc/easypdkprog/default.nix
Comment (danieldk) [inline pkgs/development/tools/misc/easypdkprog/default.nix:20]:
```suggestion
  installPhase = ''
    install -Dm755 -t $out/bin easypdkprog
  '' + lib.optionalString stdenv.isLinux ''
    install -Dm444 -t $out/etc/udev/rules.d Linux_udevrules/70-stm32vcp.rules
  '';
```

**Review by danieldk — CHANGES_REQUESTED** [review]
Starts to look good! I added another suggestion to merge `postInstall` (which was currently not working) and `installPhase`. `postInstall` is typically only useful when you use an unchanged `installPhase`.

Could you split the change where you add yourself to `maintainers.nix` to a separate commit with the following message?

```
maintainers: add david-sawatzke
```

This commit should be ordered before the commit where you add the derivation.

Let me know if you need any help!

**Comment by david-sawatzke** [comment]
Thanks, regarding:
> merge postInstall (which was currently not working)

It seems to work now after being merged, but I don't understand why it didn't work before. Do you have any idea why that's the case?

**Comment by danieldk** [comment]
> It seems to work now after being merged, but I don't understand why it didn't work before. Do you have any idea why that's the case?

`preInstall` and `postInstall` are run as hooks in `installPhase`. The default `installPhase` looks like this:

```
installPhase() {
    runHook preInstall

    # Do the installPhase stuff

    runHook postInstall
}
```

So if you override `installPhase` they are not run anymore.

**Review by danieldk — APPROVED** [review]
Looks good now, thanks a lot!

Result of `nixpkgs-review pr 91671` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
<br>- easypdkprog
</details>

**Pull request merged**
